client/Assets/Scripts/UI/Settings/ToggleButton.cs

```diff
@@ -84,6 +86,26 @@ public void ToggleClientPredictionGhost()
         ToggleUIState(battle.showClientPredictionGhost);
     }
 
+    public void ToggleZone()
+    {
+        bool zoneActive = battle.zoneActive;
+        if (zoneActive)
+        {
+            if (textState != null)
+            {
+                textState.text = "On";
+            }
+        }
+        else
+        {
+            if (textState != null)
+            {
+                textState.text = "Off";
+            }
+        }
+        ToggleUIState(zoneActive);
+    }
```

Comment:
You can rewrite this if to:
```suggestion
        public void ToggleZone()
    {
        bool zoneActive = battle.zoneActive;
        if (textState != null) {
            if(zoneActive){
                textState.text = "On";
            } else {
                textState.text = "Off";
            }
        }
        ToggleUIState(zoneActive);
    }
```

Reply:
I could change this but I'll delete it in a future PR.